### ISSUE TITLE
MySQL event publication registry schema leads to SQLException

### ISSUE THREAD

**Issue by danstooamerican** [spring-projects-experimental/spring-modulith#94]
Hi, 

I have tried to work with TransactionalEventListener but always received the SQLException `Incorrect string value: '\xA7IL\x9D\xC2\x88...' for column 'id' at row 1` whenever the handler received an event.

I am working with MySQL and initialized my schema with the following command that can be found in the [documentation](https://docs.spring.io/spring-modulith/docs/current/reference/html/#appendix.schemas) (slightly adapted because I use the JPA starter).

```
create table jpa_event_publication (
    id VARCHAR(36) not null, 
    completion_date TIMESTAMP(6), 
    event_type varchar(512), listener_id varchar(512), 
    publication_date TIMESTAMP(6), 
    serialized_event varchar(4000), 
    primary key (id)
);
```

After digging deeper into the stack trace, I discovered that the prepared statement looks like this:

```
insert into jpa_event_publication (completion_date, event_type, listener_id, publication_date, serialized_event, id) values (..., x'dfa3fa1818e447c59e214abf60964755');
```
The `x` before the actual id is not a typo and actually causes the error.

I was able to resolve the error by changing the type of the id column in the database to `BINARY(16)` as is suggested [here](https://phauer.com/2016/uuids-hibernate-mysql/).

```
create table jpa_event_publication (
    id BINARY(16) not null, 
    completion_date TIMESTAMP(6), 
    event_type varchar(512), listener_id varchar(512), 
    publication_date TIMESTAMP(6), 
    serialized_event varchar(4000), 
    primary key (id)
);
```

**Comment by danstooamerican** [comment]
Yes, this is still an issue in version 0.6.0-SNAPSHOT. I created a [reproducer](https://github.com/danstooamerican/modulith-reproducer-issue-94) which includes a failing test case.

I used test containers to spin up a MySQL database for the test. The schema is initialized with Flyway.

**Comment by ifr1m** [comment]
@danstooamerican Thanks for providing a way to reproduce the issue.

I checked and the issue is not fixed in the latest version at the moment (1.0.2).

The problem is that the JPA provider (hibernate in this case) prefers to covert the [UUID id column (from JpaEventPublication)](https://github.com/spring-projects/spring-modulith/blob/9d30fcd29c59d3f7205152a369d266757f328337/spring-modulith-events/spring-modulith-events-jpa/src/main/java/org/springframework/modulith/events/jpa/JpaEventPublication.java#L39) to binary instead of char.  

For the jdbc variant of the event publication the conversion is handled explicitly [here](https://github.com/spring-projects/spring-modulith/blob/9d30fcd29c59d3f7205152a369d266757f328337/spring-modulith-events/spring-modulith-events-jdbc/src/main/java/org/springframework/modulith/events/jdbc/DatabaseType.java#L35-L35),  but for JPA we need a different approach. 

Options:

1. In your project configure  the JPA  provider in such a way it prefers CHAR when persisting UUID values. For a setup using SpringBoot and Hibernate this can be done with `spring.jpa.properties.hibernate.type.preferred_uuid_jdbc_type=CHAR`
2. Open a PR in spring-modulith project to use `@jakarta.persistence.Convert` and a UUID converter for id column of JpaEventPublication. (this is a bit tricky as the `@Convert` does not work on `@Id` annotated columns - can be sidesteped using an `@IdClass` - will experiment to see if current tests pass using @IdClass and if there are no side effects using  an @IdClass)
3. If possible don't use the jpa variant for event publication and use the jdbc variant: spring-modulith-events-jdbc/spring-modulith-starter-jdbc

**Comment by raedbh** [comment]
It's specifically related to how [Hibernate maps the UUID type to the SQL type](https://docs.jboss.org/hibernate/orm/6.0/migration-guide/migration-guide.html#_uuid_mapping_changes).

Setting `spring.jpa.properties.hibernate.type.preferred_uuid_jdbc_type` to `VARCHAR` should fix the issue.
